### ISSUE TITLE
Z-Image transformer: SHLO→TTIR failure on RoPE complex<f32> legalization

### ISSUE THREAD

**Issue by ashokkumarkannan1** [tenstorrent/tt-xla#4756]
### Summary

- `tests/torch/models/z_image/test_transformer.py` fails at **compile** for `ZImageTransformer2DModel` (~**6.155B** params, under 7B single-chip weight budget).
 - Failure is **SHLO→TTIR legalization** around RoPE (`complex<f32>` ↔ `f32`), not weight OOM. (Log - [transformer_tt.log](https://github.com/user-attachments/files/27959684/transfortmer_t.log))

 
### Model / test
- Repo: Tongyi-MAI/Z-Image — transformer/ (ZImageTransformer2DModel)
- Bring-up: third_party/tt_forge_models/z_image — ModelVariant.TRANSFORMER, single chip, bf16
- Inputs: latents (1, 16, 160, 90), pipeline-tokenized caption embeddings, normalized timestep

### Error
`loc("p1.5"): error: failed to legalize unresolved materialization from ('tensor<512x24x2xf32>') to ('tensor<512x24xcomplex<f32>>') that remained live after conversion`

### Root cause
- ZImageTransformer2DModel fails at compile , reduced the failure to one frontend op: `gather/index on a complex64 RoPE table (the torch.polar output)`. torch.polar / precompute is not in the failing graph when the table is a lifted buffer.

Op | TT compile
-- | --
torch.polar / precompute_freqs_cis steps | OK in isolation (or separate mlir path)
table[pos_ids[:, i]] on complex64 table | Fails
torch.cat on gathered complex slices | Also fails (secondary; gather alone is enough)

Minimal repro: [512, 24] complex64 polar table (axis 1) + pos_ids[:, 1] gather → mlir e.g. 3616×24 complex ↔ f32.

Full model still fails in RopeEmbedder during _prepare_sequence (transformer_z_image.py ~L339–356, ~L784).

```
Single-chip: loc("p1.5"): 512×24 2×f32 → complex<f32>
Sharded: loc("concatenate.119"): 3616×24 complex ↔ f32
```

### Steps to Reproduce the Issue

```
git checkout akannan/zimage_shlo_bug

# Minimal: complex gather only (test expects compile failure → PASS)
python -m pytest -svv tests/torch/models/z_image/test_rope_embedder_op_sanity.py \
  -k gather_complex_polar_table_only

# Per-op bisect (steps 1–8)
python -m pytest -svv tests/torch/models/z_image/test_rope_embedder_op_sanity.py

# Standalone RoPE (no checkpoint)
python -m pytest -svv tests/torch/models/z_image/test_rope_repro_standalone.py -k compile_fails

# Full model (loads weights; xfail)
python -m pytest -svv tests/torch/models/z_image/test_transformer.py::test_transformer
```

**Comment by ashokkumarkannan1** [comment]
this PR (https://github.com/tenstorrent/tt-mlir/pull/8637) fixes the issues
